File: 0942-di-string-match/0942-di-string-match.java
class Solution {
    public int[] diStringMatch(String s) {
        int l = 0;
        int r = s.length();
        int[] res = new int[s.length()+1];
        for (int i = 0; i < s.length();i++){
            res[i] = s.charAt(i) == 'I' ? l++ : r--;
        }
        res[s.length()] = r;
        return res;
    }
}

File: 0942-di-string-match/0942-di-string-match.py
class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        l = 0
        r = len(s)
        res = []
        for char in s:
            if char == "I":
                res.append(l)
                l+=1
            else:
                res.append(r)
                r-=1
        res.append(r)
        return res